components/gist/gist.app.mjs

```diff
@@ -0,0 +1,174 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "gist",
+  propDefinitions: {
+    contactId: {
+      label: "Contact Id",
+      description: "The Id of the contact that will be retrieved",
+      type: "integer",
+      async options({
+        page, tagId = null,
+      }) {
+        if (tagId) tagId = JSON.parse(tagId).id;
+        const { contacts } = await this.listContacts({
+          params: {
+            page: page + 1,
+            tag_id: tagId,
+          },
+        });
+
+        return contacts.map(({
+          name, email, id,
+        }) => ({
+          label: `${name} - ${email}`,
+          value: id,
+        })) || [];
+      },
+    },
+    tagId: {
+      label: "Tag",
+      description: "The tag that will be added",
+      type: "any",
+      async options({ page }) {
+        const { tags } = await this.listTags({
+          params: {
+            page: page + 1,
+          },
+        });
+
+        return tags.map(({
+          name, id,
+        }) => ({
+          label: name,
+          value: JSON.stringify({
+            id,
+            name,
+          }),
+        })) || [];
+      },
+    },
```

Comment:
This seems like a good opportunity to use `withLabel`. If you set `withLabel` to true, the prop value will contain `tagId.value` and `tagId.label`
```suggestion
    tagId: {
      label: "Tag",
      description: "The tag that will be added",
      type: "integer",
      async options({ page }) {
        const { tags } = await this.listTags({
          params: {
            page: page + 1,
          },
        });

        return tags.map(({
          name, id,
        }) => ({
          label: name,
          value: id,
        })) || [];
      },
      withLabel: true,
    },
```